.github/workflows/ci.yml

```diff
@@ -356,43 +393,91 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest]
-        scala: [2.12.18, 2.13.11]
+        scala: [3, 2.12, 2.13]
         java: [temurin@11]
     runs-on: ${{ matrix.os }}
     steps:
       - name: Checkout current branch (fast)
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
-      - name: Download Java (temurin@17)
-        id: download-java-temurin-17
+      - name: Setup Java (temurin@17)
+        id: setup-java-temurin-17
         if: matrix.java == 'temurin@17'
-        uses: typelevel/download-java@v2
+        uses: actions/setup-java@v4
         with:
           distribution: temurin
           java-version: 17
+          cache: sbt
+
+      - name: sbt update
+        if: matrix.java == 'temurin@17' && steps.setup-java-temurin-17.outputs.cache-hit == 'false'
+        run: sbt +update
+
+      - run: sbt '++ ${{ matrix.scala }}' coverage rootJVM/test coverageAggregate
+
+      - uses: codecov/codecov-action@v3
+        with:
+          flags: ${{matrix.scala}}
+
+  site:
+    name: Generate Site
+    strategy:
+      matrix:
```

Comment:
Given this matrix, could the steps below be streamlined? There are checks for various Java versions, but it looks like they're unneccesary if we've picked `temurin@11` for this job?

Reply:
I tried to change `githubWorkflowJavaVersions` for just the `docs` project and run `sbt githubWorkflowGenerate` and that didn't produce a diff.

Reply:
> There are checks for various Java versions, but it looks like they're unneccesary if we've picked `temurin@11` for this job?

@adamnfish this is something that we could fix in sbt-typelevel. The reason there are checks for all the Java versions is because the setup steps for the docs is inherited from the setup steps for the primary CI jobs which _do_ need all those Java versions. This inheritance is useful in some cases the setup does other things (e.g. install native dependencies or setup a database/server) that need to be shared across the CI and docs jobs.